hibernate-core/src/main/java/org/hibernate/resource/beans/container/spi/AbstractCdiBeanContainer.java

```diff
@@ -23,35 +23,50 @@
  * @author Steve Ebersole
  */
 public abstract class AbstractCdiBeanContainer implements CdiBasedBeanContainer {
-	private Map<String,ContainedBeanImplementor<?>> beanCache = new HashMap<>();
-	private List<ContainedBeanImplementor<?>> registeredBeans = new ArrayList<>();
+	private final Map<String,ContainedBeanImplementor<?>> beanCache = new HashMap<>();
+	private final List<ContainedBeanImplementor<?>> registeredBeans = new ArrayList<>();
+
+	// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+	// un-named bean support
 
 	@Override
-	public <B> ContainedBean<B> getBean(
+	public final <B> ContainedBean<B> getBean(
 			Class<B> beanType,
 			LifecycleOptions lifecycleOptions,
 			BeanInstanceProducer fallbackProducer) {
+		return getBean( beanType, lifecycleOptions, fallbackProducer, false );
```

Comment:
This changes the default behavior, though... Including for Hibernate Search in the "delayed" case. I'm not sure of the implications.

Since you're (likely) always going to call `getBean()` with the explicit `cdiRequiredIfAvailable` flag yourself, can you please preserve the legacy behavior for this method? Just for backwards compatibility of integrations?